src/core/store/common/tokensByAsset.selectors.ts

```diff
@@ -71,6 +83,19 @@ export const selectWithdrawTokenOptionsByAsset = createSelector(
           const allowancesMap = userTokensAllowancesMap[address] ?? {};
           return createToken({ tokenData, userTokenData, allowancesMap });
         });
-      return tokens;
+      return tokens.filter((token) =>
+        isSupportedToken({ assetData, token, mainVaultToken, zapperDisabled, zapType: 'zapOutWith' })
+      );
```

Comment:
How about something like:
```suggestion
      return tokens.filter((token) => {
         const isZappable = token.supported[assetData.metadata.zapOutWith];
         return isZappable || token.address === mainVaultToken;
      }
      );
```

Reply:
Refactored here https://github.com/yearn/yearn-finance-v3/pull/691/commits/58ab9e2b15f782a16eb0ccee76556690f4f62323 so we can have all the logic in one place